helix-view/src/theme.rs

```diff
@@ -105,16 +182,16 @@ pub struct Theme {
     highlights: Vec<Style>,
 }
 
-impl<'de> Deserialize<'de> for Theme {
-    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
-    where
-        D: Deserializer<'de>,
-    {
+impl From<Value> for Theme {
+    fn from(value: Value) -> Self {
         let mut styles = HashMap::new();
         let mut scopes = Vec::new();
         let mut highlights = Vec::new();
 
-        if let Ok(mut colors) = HashMap::<String, Value>::deserialize(deserializer) {
+        let theme_values: Result<HashMap<String, Value>> =
+            toml::from_str(&value.to_string()).context("Failed to load theme");
+
+        if let Ok(mut colors) = theme_values {
```

Comment:
```
HashMap::<String, Value>::deserialize(value)
```

Should still work since `toml::Value` implements `Deserialize`.

Reply:
It does for a `Deserialize` trait implementation, see the answer below.